envoy: use errors.Is(..., net.ErrClosed) instead of string matching

tklauser opened the pull request:
Instead of comparing error strings use `errors.Is` to check whether `err` or
some error that it wraps is `net.ErrClosed` (available since Go 1.16, see
https://golang.org/pkg/net/#ErrClosed).

Same for `syscall.EINVAL`.

EDIT: pushed an additional commit to bump the `golangci-lint` GH action so it supports Go 1.16.

sayboras reviewed (approved):
Seems a good candidate here as well https://github.com/cilium/cilium/blob/2df25bb53a211ddc76349004185952783786c4c0/proxylib/test/accesslog_server.go#L106

tklauser committed:
.github/workflows: update golangci-lint GitHub action for Go 1.16 support

Bump the GH action and golangci-lint to latest versions for Go 1.16
support. Disable installation of Go from the GH action and reuse the one
already provided.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
envoy, proxylib: use errors.Is(..., net.ErrClosed) instead of string matching

Instead of comparing error strings use errors.Is to check whether err or
some error that it wraps is net.ErrClosed (available since Go 1.16, see
https://golang.org/pkg/net/#ErrClosed).

Same for syscall.EINVAL.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
